### PR TITLE
[ MB-12694 ] Adding audit user to request context

### PR DESCRIPTION
**Pull request by mmayo-truss**
Changes include:
- creating functions to insert and extract the audit user into/from the request context
- creating middleware to add audit user to request context
- modifying client cert middleware to insert user in request context instead of creating mock session
- modifying auditable app context to retrieve audit user from request context
- modifying audit logging to retrieve audit user from request context

## Summary

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Verify that changes made to the move are still documenting the user correctly in the modfied by column of the move history table
2. This needs to be verified as well in the prime api, using postman.  

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

### CODE REVIEW

**Review by kctruss — COMMENTED** [review]
@mmayo-truss I see some discrepancies between an edit made from the Prime simulator UI and via the Prime API in Postman - specifically, the Prime sim UI is not getting stamped with the "Prime" name, but instead looks like an office user.

It _is_ possible that I got my local env in a funky state from my testing, so I'd love a second set of eyes on this. @michaelvaldes-truss @ruizajtruss @afoon 
Screen Shot 2022-12-05 at 1 25 49 PM

File: pkg/services/audit/audit.go [hunk 1]
```diff
@@ -103,6 +104,17 @@ func CaptureAccountStatus(appCtx appcontext.AppContext, model interface{}, activ
 	return logItems, nil
 }
 
+func extractAuditUser(request *http.Request) []zap.Field {
+	var logItems []zap.Field
+	auditUser := audit.RetrieveAuditUserFromContext(request.Context())
+	logItems = append(logItems,
+		zap.String("audit_user_id", auditUser.ID.String()),
+		zap.String("audit_user_email", auditUser.LoginGovEmail),
+		zap.String("BLAHBLAHBLAH", "BLAH"),
```
[the remark this answers is not in the document]
Comment (michaelvaldes-truss) [reply pkg/services/audit/audit.go:113]: I can speak to the prime simulator UI part -- it is intentional. Prime simulator users are office users that are logged in and have a session, so they're treated like other office users.
Prime API calls, however, should be logged as "Prime" in the move history.

[the remark this answers is not in the document]
Comment (kctruss) [reply pkg/services/audit/audit.go:113]: @michaelvaldes-truss 🤦 I thought I remembered we overwrote the name as "Prime" for UI users, but I very definitely may have dreamed that. So the screenshot I shared looks correct to you?

**Review by kctruss — APPROVED** [review]
**praise:** Excellent job on this, @mmayo-truss ! And huge thanks to you and @michaelvaldes-truss for correcting my assumption/poor memory re: prime UI.

File: pkg/services/audit/audit.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelvaldes-truss) [reply pkg/services/audit/audit.go]: Ooh, if we overwrote that then I either didn't know about it or forgot. 
As far as I know, it looks right; this is a result of how we pull the data in the fetcher query.

**Pull request merged**